crates/core/app/tests/mock_consensus.rs

```diff
@@ -204,3 +206,30 @@ async fn mock_consensus_can_spend_notes_and_detect_outputs() -> anyhow::Result<(
 
     Ok(())
 }
+
+#[tokio::test]
+#[ignore = "this test is not currently expected to pass"] /*TODO(kate)*/
+async fn mock_consensus_view_server_test() -> anyhow::Result<()> {
+    use camino::Utf8PathBuf;
+
+    // Install a test logger, acquire some temporary storage, and start the test node.
+    let guard = common::set_tracing_subscriber();
+    let tempdir = tempdir()?;
+    let storage = TempStorage::new().await?;
+    let _test_node = common::start_test_node(&storage).await?;
+
+    const VIEW_FILE_NAME: &str = "pcli-view.sqlite"; /*copied from pcli*/
+    let url = "http:127.0.0.1:8080".parse::<url::Url>()?;
+    let sqlite_path = tempdir.path().join(VIEW_FILE_NAME);
+    let view_service = penumbra_view::ViewServer::load_or_initialize(
+        Some(Utf8PathBuf::from_path_buf(sqlite_path).unwrap()),
+        &*test_keys::FULL_VIEWING_KEY,
+        url,
+    )
+    .await?;
```

Comment:
(hacky code disclaimer) :hammer: 

this currently fails. to my understanding this is because internally, this calls into `Storage::load_or_initialize`, which itself invokes `QueryServiceClient::connect`, here:

https://github.com/penumbra-zone/penumbra/blob/a598c12ebf6f2337e597a5d79f16cbaf605f31be/crates/view/src/storage.rs#L89-L94

...which wants to connect to a `AppQueryServiceServer`. currently, that lives in `pd`, here:

https://github.com/penumbra-zone/penumbra/blob/a598c12ebf6f2337e597a5d79f16cbaf605f31be/crates/bin/pd/src/main.rs#L196-L198

i believe the correct choice forward will involve hoisting the `grpc_server` creation out of `pd` to a crate we can invoke here, does that sound correct to you as well @hdevalence?

Reply:
<question removed, i have answered my own question.>

Reply:
this led to #3980 :cherries: